pkg/lib/config/configsource/store.go

```diff
@@ -35,3 +74,85 @@ func (s *Store) GetAppIDByDomain(domain string) (string, error) {
 
 	return appID, nil
 }
+
+func (s *Store) GetDatabaseSourceByAppID(appID string) (*DatabaseSource, error) {
+	builder := s.selectConfigSourceQuery()
+	builder = builder.Where("app_id = ?", appID)
+
+	scanner, err := s.SQLExecutor.QueryRowWith(builder)
+	if err != nil {
+		return nil, err
+	}
+
+	dbs, err := s.scanConfigSource(scanner)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, ErrAppNotFound
+		}
+		return nil, err
+	}
+
+	return dbs, nil
+}
+
+func (s *Store) CreateDatabaseSource(dbs *DatabaseSource) error {
+	data, err := json.Marshal(dbs.Data)
```

Comment:
How does `data` look like in JSON? Why am I asking this is because data is of type `map[string][]byte`. JSON does not have native byte array type.

Reply:
The data is the same as format as the secret data in k8s. JSON key is the filename and the value is base64 encoded file content.